### ISSUE TITLE
Timeout doesn't work properly on windows

### ISSUE THREAD

**Issue by vinxzhu** [bugst/go-serial#130]
The ReadIntervalTimeout  on windows platform is set to -1, which causes the timeout function to work improperly.
I tried to set it to 1 in serial_windows.go, then the function worked well.
![image](https://user-images.githubusercontent.com/4533708/148159047-6f3c3186-94f5-48f6-919e-2db58b5df5d5.png)

**Comment by kroppt** [comment]
The Windows [COMMTIMEOUTS documentation](https://docs.microsoft.com/en-us/windows/win32/api/winbase/ns-winbase-commtimeouts) for defines the structure:
```c
typedef struct _COMMTIMEOUTS {
  DWORD ReadIntervalTimeout;
  DWORD ReadTotalTimeoutMultiplier;
  DWORD ReadTotalTimeoutConstant;
  DWORD WriteTotalTimeoutMultiplier;
  DWORD WriteTotalTimeoutConstant;
} COMMTIMEOUTS, *LPCOMMTIMEOUTS;
```

Again, to clarify the data type, the [data types documentation](https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types) defines:
```c
typedef unsigned long DWORD;
```

This is a uint32, which is how it is defined in the Go code as well.

Thus, the value of **ReadIntervalTimeout** cannot be -1.

And finally, the [COMMTIMEOUTS documentation](https://docs.microsoft.com/en-us/windows/win32/api/winbase/ns-winbase-commtimeouts#remarks) remarks:
> If an application sets **ReadIntervalTimeout** and **ReadTotalTimeoutMultiplier** to **MAXDWORD** and sets **ReadTotalTimeoutConstant** to a value greater than zero and less than **MAXDWORD**, one of the following occurs when the [ReadFile](https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-readfile) function is called:
>
> - If there are any bytes in the input buffer, [ReadFile](https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-readfile) returns immediately with the bytes in the buffer.
> - If there are no bytes in the input buffer, [ReadFile](https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-readfile) waits until a byte arrives and then returns immediately.
> - If no bytes arrive within the time specified by **ReadTotalTimeoutConstant**, [ReadFile](https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-readfile) times out.

In short, the code currently configures **ReadIntervalTimeout** and **ReadTotalTimeoutMultiplier** correctly, because that is the behavior we want.

I'm not saying there isn't a bug per se, but the explanation you are offering for why appears incorrect. Do you have a concise code sample that reproduces the timeout problem? What is the behavior you are seeing that makes you think it is working improperly?
